library/alloc/src/collections/btree/map.rs

```diff
@@ -355,6 +355,30 @@ pub struct ValuesMut<'a, K: 'a, V: 'a> {
     inner: IterMut<'a, K, V>,
 }
 
+/// An owning iterator over the keys of a `BTreeMap`.
+///
+/// This `struct` is created by the [`into_keys`] method on [`BTreeMap`].
+/// See its documentation for more.
+///
+/// [`into_keys`]: BTreeMap::into_keys
+#[unstable(feature = "map_into_keys_values", issue = "55214")]
+#[derive(Debug)]
+pub struct IntoKeys<K, V> {
+    inner: IntoIter<K, V>,
```

Comment:
Keeping `IntoIter` inside the `IntoKeys` and `IntoValues` structs. Implementation is pretty similar to what `Keys` and `Values` structs do.